app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -102,12 +102,12 @@ def realm_field(f, can_override = false, override = false)
 
   def interesting_klasses(obj)
     classes    = obj.all_puppetclasses
-    classes_ids = classes.reorder('').pluck('puppetclasses.id')
+    classes_ids = (classes.reorder('').pluck('puppetclasses.id') + obj.puppetclasses.map(&:id)).uniq.sort
```

Comment:
On Sep 9, 2015 10:51, "xprazak2" <notifications@github.com> wrote:
>
> In app/helpers/hosts_and_hostgroups_helper.rb:
>
> > @@ -102,12 +102,12 @@ def realm_field(f, can_override = false, override
= false)
> >
> >    def interesting_klasses(obj)
> >      classes    = obj.all_puppetclasses
> > -    classes_ids = classes.reorder('').pluck('puppetclasses.id')
> > +    classes_ids = (classes.reorder('').pluck('puppetclasses.id') +
obj.puppetclasses.map(&:id)).uniq.sort
>
> @ohadlevy: for the same reason we call reorder? I am convinced we do not
need it but there must be some reason it is there, perhaps I just do not
see it...
>

I think we remove ordering to make the db query faster as we don't care
about ordering by name for example.
Further pg will complain if you order by a field which is not in the select
statement.

> —
> Reply to this email directly or view it on GitHub.